File: elixir/roman_numerals.ex
defmodule RomanNumerals do
  @doc """
  Convert the number to a roman number.
  """
  @spec numeral(pos_integer) :: String.t()
  def numeral(number) do
    letters = [~w(I V X), ~w(X L C), ~w(C D M), ~w(M M M)]
    aux(number, letters, [])
  end

  defp aux(0, _, ans) do
    ans |> Enum.join()
  end

  defp aux(n, [[l, v, x] | ls], ans) do
    d = rem(n, 10)

    ans =
      cond do
        d == 0 ->
          ans

        d in 1..3 ->
          [String.duplicate(l, d) | ans]

        d == 4 ->
          [l <> v | ans]

        d == 5 ->
          [v | ans]

        d in 6..8 ->
          [v <> String.duplicate(l, d - 5) | ans]

        9 ->
          [l <> x | ans]
      end

    aux(div(n, 10), ls, ans)
  end
end

File: python/roman_numerals.py
def roman(number):
    letters = [('I', 'V', 'X'), ('X', 'L', 'C'), ('C', 'D', 'M'),
               ('M', 'M', 'M')]

    res = ''
    while number > 0:
        (l, v, x), *rest = letters
        d = number % 10
        if d == 0:
            pass
        elif d in [1, 2, 3]:
            res = l * d + res
        elif d == 4:
            res = l + v + res
        elif d == 5:
            res = v + res
        elif d in [6, 7, 8]:
            res = v + l * (d - 5) + res
        else:
            res = l + x + res
        letters = rest
        number //= 10
    return res
